src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1015,3 +1065,923 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
+
+    # Posting hybrid connection details to proxy in order to get kubeconfig
+    response = make_api_call_with_retries(uri, data, False, consts.Post_Hybridconn_Fault_Type,
+                                          'Unable to post hybrid connection details to clientproxy',
+                                          "Failed to pass hybrid connection details to proxy.")
+
+    # Decoding kubeconfig into a string
+    try:
+        kubeconfig = json.loads(response.text)
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Load_Kubeconfig_Fault_Type,
+                                summary='Unable to load Kubeconfig')
+        raise CLIError("Failed to load kubeonfig." + str(e))
+
+    kubeconfig = kubeconfig['kubeconfigs'][0]['value']
+    kubeconfig = b64decode(kubeconfig).decode("utf-8")
+
+    try:
+        print_or_merge_credentials(path, kubeconfig, overwrite_existing, context_name)
+        print("You can now start sending requests using kubectl on the current context.")
+
+    except Exception as e:
+        telemetry.set_exception(exception=e, fault_type=consts.Merge_Kubeconfig_Fault_Type,
+                                summary='Unable to merge kubeconfig.')
+        raise CLIError("Failed to merge kubeconfig." + str(e))
+
+
+def make_api_call_with_retries(uri, data, tls_verify, fault_type, summary, cli_error):
+    for i in range(consts.API_CALL_RETRIES):
+        try:
+            response = requests.post(uri, json=data, verify=tls_verify)
+            return response
+        except Exception as e:
+            if i != consts.API_CALL_RETRIES - 1:
+                pass
+            else:
+                telemetry.set_exception(exception=e, fault_type=fault_type, summary=summary)
+                raise CLIError(cli_error + str(e))
+
+
+def insert_token_in_kubeconfig(data, token):
+    b64kubeconfig = data['kubeconfigs'][0]['value']
+    decoded_kubeconfig_str = b64decode(b64kubeconfig).decode("utf-8")
+    dict_yaml = yaml.safe_load(decoded_kubeconfig_str)
+    dict_yaml['users'][0]['user']['token'] = token
+    kubeconfig = yaml.dump(dict_yaml).encode("utf-8")
+    b64kubeconfig = b64encode(kubeconfig).decode("utf-8")
+    return b64kubeconfig
+
+
+def check_process(processName):
+    '''
+    Check if there is any running process that contains the given name processName.
+    '''
+    for proc in process_iter():
+        try:
+            if proc.name().startswith(processName):
+                return True
+        except (NoSuchProcess, AccessDenied, ZombieProcess):
+            pass
+    return False
+
+
+def ctrlc_handler(sig, frame):
+    interrupt_main()
+
+
+def get_custom_locations_oid(cmd):
+    try:
+        sp_graph_client = get_graph_client_service_principals(cmd.cli_ctx)
+        sub_filters = []
+        sub_filters.append("startswith(displayName,'{}')".format("Custom Locations RP"))
```

Comment:
Can we use the appId instead of name to filter the oid?

Reply:
@atharvamulmuley on this?

Reply:
@akashkeshari I think its better to use name as chances of name getting changed is less than that of appid.